app/payment-request/index.js

```diff
@@ -0,0 +1,43 @@
+const db = require('../data')
+const paymentRequestSchema = require('./payment-request-schema')
+
+async function savePaymentRequest (paymentRequest) {
+  const validationResult = paymentRequestSchema.validate(paymentRequest)
+
+  if (validationResult.error) {
+    throw new Error(`Payment request is invalid. ${validationResult.error.message}`)
+  }
+
+  await db.paymentRequest.create({
+    sourceSystem: paymentRequest.sourceSystem,
+    deliveryBody: paymentRequest.deliveryBody,
+    invoiceNumber: paymentRequest.invoiceNumber,
+    frn: paymentRequest.frn,
+    sbi: paymentRequest.sbi,
+    paymentRequestNumber: paymentRequest.paymentRequestNumber,
+    agreementNumber: paymentRequest.agreementNumber,
+    contractNumber: paymentRequest.contractNumber,
+    currency: paymentRequest.currency,
+    schedule: paymentRequest.schedule,
+    dueDate: paymentRequest.dueDate,
+    value: paymentRequest.value
+  }).then(result => processInvoiceLines(paymentRequest.invoiceLines, result.paymentRequestId))
+}
```

Comment:
As all the properties are the same for each object you can simplify this to just pass the object rather than list every field.

eg.

`await db.create(paymentRequest)` would achieve the same result.`

Also you've got a mix of `await` and `.then` approach for handling promises.  As we're using `await` everywhere else, I would change this to:

```
const savedPaymentRequest = await db.paymentRequest.create(paymentRequest)
await processInvoiceLines(invoiceLines, savedPaymentRequest.paymentRequestId
```

Reply:
agree